src/lib/logger.ts

```diff
@@ -52,8 +55,168 @@ const browserLoggerFn: LoggerFunction = (level, data) => {
   )
 }
 
-export const logger = new Logger({
-  name: 'App',
-  level: getConfig().REACT_APP_LOG_LEVEL,
-  logFn: browserLoggerFn,
-})
+/**
+ * Get the current log level configuration from local storage or from the environment
+ *
+ * We can't get the log level from the logger instance because it's stored in a private
+ * variable that contains a numeric value.
+ */
+export const getLogLevel = () => {
+  try {
+    return globalThis.localStorage?.getItem?.('LOG_LEVEL') || getConfig().REACT_APP_LOG_LEVEL
+  } catch (e) {
+    return 'debug'
+  }
+}
+
+/**
+ * Instantiate a new Logger with some default options
+ */
+export const createLogger = (opts?: LoggerOptions) => {
+  const options = {
+    name: 'App',
+    logFn: browserLoggerFn,
+    level: getLogLevel(),
+    ...opts,
+  }
+  const l = new Logger(options)
+  l.debug({ options }, 'New Logger Created')
+  return l
+}
+
+const LogLevelStrings = Object.values<string>(LogLevel)
+const isValidLogLevel = (level: string): level is LogLevel => {
+  return LogLevelStrings.includes(level)
+}
+
+/**
+ * Save the specified LogLevel to LocalStorage
+ */
+export const saveLogLevel = (level: LogLevel | string) => {
+  if (isValidLogLevel(level)) {
+    try {
+      globalThis.localStorage?.setItem?.('LOG_LEVEL', level)
+      return true
+    } catch (e) {
+      logger.error(e, { fn: 'saveLogLevel', level }, 'Error persisting LOG_LEVEL to LocalStorage')
+    }
+  }
+
+  return false
+}
+
+export const logger = createLogger()
+
+/*
+ DOCUMENATION
+
+ # Why
+ The purpose of this logger is to allow us to log out information in a structured way that
+ makes debugging easier by allowing us to more easily filter and search logs.
```

Comment:
Great docs, though I feel like they should be in a README somewhere?

Reply:
I asked def1 about this and he suggested that most developers would come across this code and its comment before looking at a file somewhere else. 

I will probably also make a PR to the Docs repo. If so I'll link that from here.